NO-JIRA: Don't run unit tests again for SonarQube

thahnen opened the pull request:
This is primarily used to not be blocked if unit tests decide to be flaky here instead of the actual unit-test-invoking "validate" task. On the other hand it speeds up the build by a minute or so.

thahnen committed:
NO-JIRA: Don't run unit tests again for SonarQube

Currently we run the unit tests in the "validate" task and again in the "sonarqube" task. This is unnecessary and sometimes makes flaky tests appear in the "sonarqube" task which is unexpected.

jblievremont commented on .cirrus.default.yml:
```diff
@@ -270,7 +270,7 @@ sonarqube_task:
     source cirrus-env QA
     .cirrus/regular_mvn_build_deploy_analyze \
-      -P-deploy-sonarsource,-release,-sign -Dmaven.install.skip=true -Dmaven.deploy.skip=true \
+      -P-deploy-sonarsource,-release,-sign -Dmaven.install.skip=true -DskipTests -Dmaven.deploy.skip=true \
```
For consistency, may I suggest using `maven.test.skip=true` 😇?

thahnen replied:
We cannot use it, as it is for Tycho Surefire, not Maven :open_mouth:

thahnen committed:
NO-JIRA: JaCoCo on unit tests earlier

On SonarQube task only load JaCoCo results from both unit and integration tests.

jblievremont reviewed (approved):
LGTM!
